Sources/Client/Client/Client+Channel.swift

```diff
@@ -43,20 +43,28 @@ public extension Client {
     ///   - type: a channel type.
     ///   - members: a list of members.
     ///   - extraData: a channel extra data.
+    ///   - namingStrategy: a namiging strategy to generate a name and image for the channel based on members.
     func channel(type: ChannelType = .messaging,
                  members: [User],
-                 extraData: ChannelExtraDataCodable? = nil) -> Channel {
-        Channel(type: type,
-                id: "",
-                members: members,
-                invitedMembers: [],
-                extraData: extraData,
-                created: .init(),
-                deleted: nil,
-                createdBy: nil,
-                lastMessageDate: nil,
-                frozen: false,
-                config: .init())
+                 extraData: ChannelExtraDataCodable? = nil,
+                 namingStrategy: ChannelNamingStrategy? = DefaultChannelNamingStrategy()) -> Channel {
```

Comment:
I think this shouldn't be a parameter, why would users need this? They can just pass the names via `extraData`

Reply:
Maybe we shouldn't expose it at all?

Reply:
It starts making sense with `maxNumberOfNames` in the initializer.
Users then can do:
```swift
Client.cannel(members: [] namingStrategy: DefaultChannelNamingStrategy(maxUserNames: 1))`
```
to get channel names like "John and 5 more".